File: app/src/main/java/com/marked/pixsee/friends/FriendsAdapter.java
package com.marked.pixsee.friends;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.marked.pixsee.R;
import com.marked.pixsee.data.user.User;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

import static com.facebook.common.internal.Preconditions.checkNotNull;


/**
 * Created by Tudor Pop on 30-Nov-15.
 */
class FriendsAdapter extends RecyclerView.Adapter<FriendVH> {
	private final List<User> mDataSet;
	private FriendFragment.FriendFragmentInteractionListener friendFragmentInteractionListener;

	FriendsAdapter(@NotNull FriendFragment.FriendFragmentInteractionListener friendFragmentInteractionListener) {
		this.friendFragmentInteractionListener = checkNotNull(friendFragmentInteractionListener);
		mDataSet= new ArrayList<>();
	}

	@Override
	public FriendVH onCreateViewHolder(ViewGroup parent, int viewType) {
		View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_friend, parent, false);

		return new FriendVH(v,friendFragmentInteractionListener);
	}

	public void setDataSet(List<User> dataSet) {
		mDataSet.clear();
		mDataSet.addAll(dataSet);
	}

	@Override
	public void onBindViewHolder(FriendVH holder, int position) {
		final User friend = mDataSet.get(position);
		holder.bindFriend(friend);
	}

	// Return the size of your mDataSet (invoked by the layout manager)

	@Override
	public int getItemCount() {
		return mDataSet.size();
	}
}

File: app/src/main/java/com/marked/pixsee/friends/FriendsAdapter.kt
package com.marked.pixsee.friends;

import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.marked.pixsee.R
import com.marked.pixsee.data.friend.Friend
import com.marked.pixsee.frienddetail.ChatDetailActivity
import java.util.*

/**
 * Created by Tudor Pop on 30-Nov-15.
 */
class FriendsAdapter(private val context: Context, private var dataSet: ArrayList<Friend>) : RecyclerView.Adapter<FriendHolder>() {

	override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FriendHolder {
        // Create a new view.
		val v = LayoutInflater.from(parent.context).inflate(R.layout.contact_item, parent, false);
		val holder = FriendHolder(v, context);
        return holder;
    }

	override fun onBindViewHolder(holder: FriendHolder, position: Int) {
        val contact = dataSet.get(position);
		holder.apply {
			bindContact(contact)
			itemView.setOnClickListener { view ->
				// itemView is a public member that points to the root element of the contactHolder (in the layout)
				// TODO: 03-Dec-15 send contact detail from here to ContactDetailActivity
				// send the clicked contact to detail activity
				val intent = Intent(context, ChatDetailActivity::class.java)
				intent.putExtra(ChatDetailActivity.EXTRA_CONTACT, contact);
                context.startActivity(intent)
//				context.startActivity(intent, ActivityOptionsCompat.makeSceneTransitionAnimation(context as ContactListActivity).toBundle()); // RIGHT !!!
			}
		};
    }

    // Return the size of your dataset (invoked by the layout manager)
    override fun getItemCount(): Int {
        return dataSet.size;
    }

}
